src/arch/ucx/machine.C

```diff
@@ -372,99 +438,37 @@ static void UcxRxReqCompleted(void *request, ucs_status_t status,
-
-        NcpyOperationInfo *ncpyOpInfo = (NcpyOperationInfo *)(req->msgBuf);
-        UCX_LOG(4, "Got ncpy size %d (meta size %d)", ncpyOpInfo->srcSize, ncpyOpInfo->ncpyOpInfoSize);
-        resetNcpyOpInfoPointers(ncpyOpInfo);
-
-        UcxRdmaInfo *info = (UcxRdmaInfo *)(ncpyOpInfo->srcLayerInfo + CmiGetRdmaCommonInfoSize());
-
-        UcxMemMap(info,
-                  (void *)ncpyOpInfo->srcPtr,
-                  ncpyOpInfo->srcSize);
-
-        ncpyOpInfo->isSrcRegistered = 1;
-
-        ncpyOpInfo->freeMe = CMK_FREE_NCPYOPINFO; // It's a message, not a realy ncpy Obj
-        UCX_LOG(4, "Reset ncpy size %d (meta size %d)", ncpyOpInfo->destSize, ncpyOpInfo->ncpyOpInfoSize);
-
-        // send back to destination process to perform GET
-        UcxSendMsg(CmiNodeOf(ncpyOpInfo->destPe), ncpyOpInfo->destPe,
-                   ncpyOpInfo->ncpyOpInfoSize, (char*)ncpyOpInfo,
-                   UCX_RMA_TAG_GET, UcxRmaSendCompletedAndFree);
-
-    } else if (info->sender_tag & UCX_RMA_TAG_GET) {
-        NcpyOperationInfo *ncpyOpInfo = (NcpyOperationInfo *)(req->msgBuf);
-        resetNcpyOpInfoPointers(ncpyOpInfo);
-
-        ncpyOpInfo->freeMe = CMK_FREE_NCPYOPINFO; // It's a message, not a real ncpy Obj
-        UcxRmaOp(ncpyOpInfo, UCX_RMA_OP_GET);
-
-    } else if (info->sender_tag & UCX_RMA_TAG_DEREG_AND_ACK) {
-        NcpyOperationInfo *ncpyOpInfo = (NcpyOperationInfo *)(req->msgBuf);
-        resetNcpyOpInfoPointers(ncpyOpInfo);
-        ncpyOpInfo->freeMe = CMK_FREE_NCPYOPINFO;
-
-        if(CmiMyNode() == CmiNodeOf(ncpyOpInfo->srcPe)) { // source node
-            LrtsDeregisterMem(ncpyOpInfo->srcPtr,
-                              ncpyOpInfo->srcLayerInfo + CmiGetRdmaCommonInfoSize(),
-                              ncpyOpInfo->srcPe,
-                              ncpyOpInfo->srcRegMode);
-
-            ncpyOpInfo->isSrcRegistered = 0; // Set isSrcRegistered to 0 after de-registration
-
-            // Invoke source ack
-            if(ncpyOpInfo->opMode != CMK_BCAST_EM_API) {
-                ncpyOpInfo->opMode = CMK_EM_API_SRC_ACK_INVOKE;
-                CmiInvokeNcpyAck(ncpyOpInfo);
-            }
-
-        } else if(CmiMyNode() == CmiNodeOf(ncpyOpInfo->destPe)) { // destination node
-
-            LrtsDeregisterMem(ncpyOpInfo->destPtr,
-                              ncpyOpInfo->destLayerInfo + CmiGetRdmaCommonInfoSize(),
-                              ncpyOpInfo->destPe,
-                              ncpyOpInfo->destRegMode);
-
-            ncpyOpInfo->isDestRegistered = 0; // Set isDestRegistered to 0 after de-registration
-
-            // Invoke destination ack
-            ncpyOpInfo->opMode = CMK_EM_API_DEST_ACK_INVOKE;
-            CmiInvokeNcpyAck(ncpyOpInfo);
-
-        } else {
-            CmiAbort(" Cannot de-register on a different node than the source or destinaton");
-        }
-    }
-#endif
-
     if (req->msgBuf != NULL) {
         // Request is not completed immideately
```

Comment:
Minor Spelling Fix (not related to this patch), but could go in this patch: immideately => immediately (has multiple occurrences in this file).

Reply:
ok